conans/server/launcher.py

```diff
@@ -15,12 +15,18 @@
 
 
 class ServerLauncher(object):
-    def __init__(self, force_migration=False):
+    def __init__(self, force_migration=False, server_dir=None):
         self.force_migration = force_migration
-        user_folder = conan_expand_user("~")
-        server_folder = os.path.join(user_folder, '.conan_server')
+        if server_dir:
+            server_folder = server_dir
+        else:
+            user_folder = conan_expand_user("~")
+            server_folder = os.path.join(user_folder, '.conan_server')
 
-        server_config = migrate_and_get_server_config(user_folder, self.force_migration)
+        is_custom_path = server_dir is not None
+        server_config = migrate_and_get_server_config(
+            user_folder, is_custom_path, self.force_migration
```

Comment:
This is calling the arguments in the wrong order

Reply:
Yikes, sorry, fixing it